v2/dtos/requests/deviceprofile.go

```diff
@@ -0,0 +1,184 @@
+func AddDeviceProfileReqToDeviceProfileModel(addReq AddDeviceProfileRequest) (DeviceProfiles models.DeviceProfile) {
+	deviceResources := make([]models.DeviceResource, len(addReq.DeviceResources))
+	for i, dr := range addReq.DeviceResources {
+		deviceResources[i] = dtos.ToDeviceResourceModel(dr)
+	}
+	deviceCommands := make([]models.ProfileResource, len(addReq.DeviceCommands))
+	for i, dc := range addReq.DeviceCommands {
+		deviceCommands[i] = dtos.ToProfileResourceModel(dc)
+	}
+	commands := make([]models.Command, len(addReq.CoreCommands))
+	for i, c := range addReq.CoreCommands {
+		commands[i] = dtos.ToCommandModel(c)
+	}
+
+	return models.DeviceProfile{
+		Name:            addReq.Name,
+		Description:     addReq.Description,
+		Manufacturer:    addReq.Manufacturer,
+		Model:           addReq.Model,
+		Labels:          addReq.Labels,
+		DeviceResources: deviceResources,
+		DeviceCommands:  deviceCommands,
+		CoreCommands:    commands,
+	}
+}
+
+// AddDeviceProfileReqToDeviceProfileModels transforms the AddDeviceProfileRequest DTO array to the DeviceProfile model array
+func AddDeviceProfileReqToDeviceProfileModels(addRequests []AddDeviceProfileRequest) (DeviceProfiles []models.DeviceProfile) {
+	for _, req := range addRequests {
+		dp := AddDeviceProfileReqToDeviceProfileModel(req)
+		DeviceProfiles = append(DeviceProfiles, dp)
+	}
+	return DeviceProfiles
+}
+
+// UpdateDeviceProfileRequest defines the Request Content for PUT event as pushed DTO.
+// This object and its properties correspond to the UpdateDeviceProfileRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/UpdateDeviceProfileRequest
+type UpdateDeviceProfileRequest struct {
+	common.BaseRequest `json:",inline"`
+	Id                 string                 `json:"id,omitempty" yaml:"id,omitempty"`
+	Name               string                 `json:"name,omitempty" yaml:"name,omitempty"`
+	Manufacturer       string                 `json:"manufacturer,omitempty" yaml:"manufacturer,omitempty"`
+	Description        string                 `json:"description,omitempty" yaml:"description,omitempty"`
+	Model              string                 `json:"model,omitempty" yaml:"model,omitempty"`
+	Labels             []string               `json:"labels,omitempty" yaml:"labels,flow,omitempty"`
+	DeviceResources    []dtos.DeviceResource  `json:"deviceResources,omitempty" yaml:"deviceResources,omitempty"`
+	DeviceCommands     []dtos.ProfileResource `json:"deviceCommands,omitempty" yaml:"deviceCommands,omitempty"`
+	CoreCommands       []dtos.Command         `json:"coreCommands,omitempty" yaml:"coreCommands,omitempty"`
+}
+
+// Validate satisfies the Validator interface
+func (dp UpdateDeviceProfileRequest) Validate() error {
+	err := v2.Validate(dp)
+	return err
+}
+
+// UnmarshalJSON implements the Unmarshaler interface for the UpdateDeviceProfileRequest type
+func (dp *UpdateDeviceProfileRequest) UnmarshalJSON(b []byte) error {
```

Comment:
also need UnmarshalYAML.??

Reply:
The Swagger file doesn't design PUT API for yaml file.

Reply:
Still need to answer if V2 should have PUT API for yaml file?

Reply:
@lenny-intel we might not have to deal with it this PR.
We can implement what we designed for now, and discuss the PUT API for yaml in the WG meeting.